Update the out-of-date readme file (#72)
canonical/prometheus-hardware-exporter: README.md

# Prometheus Hardware Exporter

Prometheus Hardware Exporter exports Prometheus metrics from BMCs (Baseboard
management controllers) using IPMI (Intelligent Platform Management Interface)
and Redfish protocol. It also exports and various SAS (Serial Attached SCSI) and

> RAID (redundant array of inexpensive disks) controllers.

Reviewer: Would be nice to add a line mentioning that it's meant to be used alongside hardware observer. The line in the next paragraph only mentions "you can learn more about hardware observer..."

Author: Done

Note that the exporter does not bundle the required third party or proprietary
software. If you would like to use this exporter you will need to install them
by yourself. Therefore, Prometheus Hardware Exporter is recommended for
[Juju](https://juju.is/) users, you can learn more about Hardware Observer on

> [Charmhub](https://charmhub.io/hardware-observer).

Author: Done

## Installation

This package is not published in anywhere yet. For now, you can should install
this application locally (or via a virtual environment). To install it on a

> virtual environment, you can run:

Author: Done

```shell
# Install virtualvenv and source it
pip3 install virtualenv
python3 -m venv venv
source venv/bin/activate

# Install the prometheus-hardware-exporter package; the "(venv)" means you are

> # under the virtual environment

Author: Done

(venv) pip3 install .
```

> To verify you have install the package, you can run:

Reviewer: ```suggestion
To verify that the package has been installed successfully, run:
```
I noticed that in some places we're referring to it as a "package" and some other places it's an "application". I don't mind the usage of either term as long as it's consistently used everywhere :smile:

Author: Done

# Prometheus Hardware Exporter

Prometheus Hardware Exporter exports Prometheus metrics from BMCs (Baseboard
management controllers) using IPMI (Intelligent Platform Management Interface)
and Redfish protocol. It also exports and various SAS (Serial Attached SCSI) and
RAID (redundant array of inexpensive disks) controllers.

> Prometheus Hardware Exporter is recommended for [Juju](https://juju.is/) users,

Reviewer 2: Non blocking. 
Another idea is that we don't need to mention the charm in exporter at all.

Author: I think this is fine to mention here, after all, this package is very hard to use without the charm to do some automation for you. And to avoid duplication of documentation effort, we need to reference to charmhub documentation

software. If you would like to use this exporter, you will need to install them
by yourself.

## Installation

This package is not published anywhere as of now. Currently, it can be installed
locally or on a virtual environment. To install it on a virtual environment,
run:

```shell
# Install virtualvenv and source it
pip3 install virtualenv
python3 -m venv venv
source venv/bin/activate

# Install the prometheus-hardware-exporter package; the "(venv)" prefix on the
# shell prompt indicates that the virtual environment is active.
(venv) pip3 install .
```

To verify that the package has been installed successfully, run:

```shell
(venv) prometheus-hardware-exporter -h
```

If you see the help message, then `prometheus-hardware-exporter` is installed on
the virtual environment.

## Usages

To start the exporter at port `10000` and enable the ipmi sensor collector, run:

```shell
(venv) prometheus-hardware-exporter -p 10000 --collector.ipmi_sensor
```

Note that all the exporters are disabled by default as you will need to install
the appropriate third party or proprietary software to run the collectors.  You

> can check out [Resources](200~https://charmhub.io/hardware-observer/resources/)

Reviewer 2: The link seems incorrect. There is a "200- " in front of the http link.

Author: Fixed :)

to find out more information about the third party software.

## Supported Metrics

> You can learn more about the metrics from [Hardware Observer

Reviewer 2: non-blocking.
I also think we should move the metrics list to here. And put the reference link in charmhub.

Author: this will be a maintenance cost: we need to maintain two copies of the same thing

Reviewer 2: There is only 1 copy. 
What I mean is to move the metrics documentation to this repository and link to here in charmhub.

Author: ah, sorry. Yeah, that makes sense

Author: The scope is bit big to do it right now, I will propose if it's of interest, we can open an single ticket to do that
